ecommerce/extensions/payment/processors/payu.py

```diff
@@ -0,0 +1,292 @@
+""" PayU payment processing. """
+# from __future__ import unicode_literals
+
+# import datetime
+# import logging
+# import uuid
+from decimal import Decimal
+from hashlib import md5
+
+# from django.conf import settings
+from oscar.apps.payment.exceptions import GatewayError, TransactionDeclined
+from oscar.core.loading import get_model
+# from suds.client import Client
+# from suds.sudsobject import asdict
+# from suds.wsse import Security, UsernameToken
+# from threadlocals.threadlocals import get_current_request
+
+from ecommerce.core.url_utils import get_lms_url
+# from ecommerce.core.constants import ISO_8601_FORMAT
+from ecommerce.extensions.order.constants import PaymentEventTypeName
+# from ecommerce.extensions.payment.constants import CYBERSOURCE_CARD_TYPE_MAP
+from ecommerce.extensions.payment.exceptions import (InvalidSignatureError, InvalidPayUDecision)
+# from ecommerce.extensions.payment.helpers import sign
+from ecommerce.extensions.payment.processors import BasePaymentProcessor
+# from ecommerce.extensions.payment.transport import RequestsTransport
+
+# logger = logging.getLogger(__name__)
+
+PaymentEvent = get_model('order', 'PaymentEvent')
+PaymentEventType = get_model('order', 'PaymentEventType')
+# PaymentProcessorResponse = get_model('payment', 'PaymentProcessorResponse')
+ProductClass = get_model('catalogue', 'ProductClass')
+Source = get_model('payment', 'Source')
+SourceType = get_model('payment', 'SourceType')
+
+
+class Payu(BasePaymentProcessor):
+    """
+    PayU payment processor (November 2016)
+
+    For reference, see
+    http://developers.payulatam.com/en/web_checkout/integration.html
+    """
+
+    NAME = 'payu'
+
+    def __init__(self):
+        """
+        Constructs a new instance of the PayU processor.
+
+        Raises:
+            KeyError: If no settings configured for this payment processor
+            AttributeError: If LANGUAGE_CODE setting is not set.
+        """
+        configuration = self.configuration
+        self.payment_page_url = configuration['payment_page_url']
+        self.merchantId = configuration['merchant_id']
```

Comment:
Later on I realized this is because the payu vars are not very standard, but we should still fix our own vars